src/hoi4_world/countries/hoi4_country_converter_tests.cpp

```diff
@@ -1550,4 +1588,105 @@ TEST(Hoi4worldCountriesCountryConverter, CharactersConvert)
                    .general_data = expected_data_three,
                }))));
 }
+
+
+TEST(Hoi4worldCountriesCountryConverter, IdeologySupportIsConverted)
+{
+   const mappers::CountryMapper country_mapper({{1, "TAG"}});
+   const vic3::Country source_country_one({.number = 1, .capital_state = 2, .ig_ids = {1, 2}});
+   std::map<int, hoi4::Character> dummy_characters;
+   std::map<std::string, mappers::CultureQueue> dummy_culture_queues;
+
+   const std::map<int, int> vic3_state_ids_to_hoi4_state_ids{{2, 4}};
+
+   const auto country_one = ConvertCountry(source_country_one,
+       {},
+       {},
+       commonItems::LocalizationDatabase{{}, {}},
+       country_mapper,
+       vic3_state_ids_to_hoi4_state_ids,
+       {},
+       mappers::IdeologyMapper(
+           {
+               {"test_law_one",
+                   {
+                       {"democratic", 2},
+                       {"fascist", 3},
+                   }},
+               {"test_law_two",
+                   {
+                       {"communist", 5},
+                       {"fascist", 3},
+                   }},
+               {"test_law_three",
+                   {
+                       {"democratic", 5},
+                       {"communist", 7},
+                   }},
+           },
+           {}),
+       {},
+       {},
+       {},
+       {},
+       {},
+       mappers::CultureGraphicsMapper{{}},
+       {},
+       mappers::LeaderTypeMapper({}),
+       mappers::CharacterTraitMapper({}, {}, {}),
+       {
+           {1, vic3::InterestGroup("test_group_one", 1, 0, 50.F, false, {"test_ideology_one", "test_ideology_two"})},
+           {2, vic3::InterestGroup("test_group_two", 1, 0, 150.F, false, {"test_ideology_one", "test_ideology_three"})},
+       },
+       vic3::Ideologies({
+           {"test_ideology_one", vic3::Ideology({{"test_law_one", 1}})},
+           {"test_ideology_two", vic3::Ideology({{"test_law_two", 2}})},
+           {"test_ideology_three", vic3::Ideology({{"test_law_three", -2}})},
+       }),
+       dummy_characters,
+       dummy_culture_queues);
+
+   ASSERT_TRUE(country_one.has_value());
+   EXPECT_THAT(country_one->GetIdeologySupport(),
+       testing::UnorderedElementsAre(testing::Pair("communist", 57),
+           testing::Pair("democratic", 42),
```

Comment:
Is the piechart supposed to hit 100 or 99? Or more likely are these raw weights?

Reply:
I just failed to round.